### PR TITLE
Facilitate erdblick web deployment

### LINKED ISSUE #102: Facilitate erdblick web deployment
As we are trying to host erdblick on viewer.nds.live, a number of tweaks need to be made to make erdblick a good internet citizen, my lowering the number of requests it makes. This includes:

* Switch to non-ESM cesium builds with inlined worker JS sources.
* Close fetch connection as soon as possible by processing the End-Of-Stream Message.
* Do not send empty or redundant /tiles fetch-operations.

### PR DESCRIPTION
**Pull request by josephbirkner**
The ESM Cesium Module, which is used by default with Typescript/Webpack, has over 100 worker source files outside of the cesium sources. Loading these creates a load of overhead and load balancing throttling on viewer.nds.live. By switching to the non-ESM sources, we can take advantage of the inlined worker sources in the Cesium.js bundle. This has the added benefit of reducing load times. See https://github.com/CesiumGS/cesium/pull/11519

Closes #102

### CODE REVIEW

File: erdblick_app/app/cesium.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// This file allows us to use ESM-Style Imports in the
+// rest of the erdblick code. However, we instruct webpack to ignore
+// the cesium import. Instead, Cesium is provided via its
+// non-ESM distribution, because it has inline workers.
+// See: https://github.com/CesiumGS/cesium/pull/11519
+// In index.js, we have the Cesium.js import, so Cesium is
+// available as a global variable.
+
+import * as Cesium from "cesium";
```
Comment (Waguramu) [inline erdblick_app/app/cesium.ts:9]: Can we come up with a better way to expose Cesium for imports?

**Review by Waguramu — APPROVED** [review]

**Comment by ashley-mort** [comment]
@josephbirkner Is there something like this that can be done to the Cesium repo build to get the workers inlined into the Cesium build that can be imported as ESM?

Apologies for the question as I'm new to the javascript bundlers and all their configurations.

**Comment by josephbirkner** [comment]
> @josephbirkner Is there something like this that can be done to the native Cesium repo build to get the workers inlined into the build that can be imported as ESM?
> 
> Apologies for the question as I'm new to the javascript bundlers and all their configurations.
> 
> I tried building this repo with `npm install` and `npm run build` but get errors like:
> 
> ./erdblick_app/app/app.component.ts:4:0-68 - Error: Module not found: Error: Can't resolve '../../build/libs/core/erdblick-core' in 'D:\CesiumTest\erdblick-switch-cesium-build\erdblick_app\app'
> 
> Error: erdblick_app/app/app.component.ts:6:70 - error TS2307: Cannot find module '../../build/libs/core/erdblick-core' or its corresponding type declarations.
> 
> 6 import MainModuleFactory, {Feature, MainModule as ErdblickCore} from '../../build/libs/core/erdblick-core'; ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Hi Ashely, yes, I think our solution will also work for you. Building Erdblick can be done by first executing `./ci/00_setup.bash`, then `./ci/10_linux_build.bash`. Erdblick uses cmake and emscripten to compile its core library to web assembly, hence the errors you are seeing.

**Comment by ashley-mort** [comment]
> > @josephbirkner Is there something like this that can be done to the native Cesium repo build to get the workers inlined into the build that can be imported as ESM?
> > Apologies for the question as I'm new to the javascript bundlers and all their configurations.
> > I tried building this repo with `npm install` and `npm run build` but get errors like:
> > ./erdblick_app/app/app.component.ts:4:0-68 - Error: Module not found: Error: Can't resolve '../../build/libs/core/erdblick-core' in 'D:\CesiumTest\erdblick-switch-cesium-build\erdblick_app\app'
> > Error: erdblick_app/app/app.component.ts:6:70 - error TS2307: Cannot find module '../../build/libs/core/erdblick-core' or its corresponding type declarations.
> > 6 import MainModuleFactory, {Feature, MainModule as ErdblickCore} from '../../build/libs/core/erdblick-core'; ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 
> Hi Ashely, yes, I think our solution will also work for you. Building Erdblick can be done by following these [instructions](https://github.com/ndsev/erdblick#build-instructions-linux-only). Erdblick uses cmake and emscripten to compile its core library to web assembly, hence the errors you are seeing.

Using cmake 3.25.3:
-- Could NOT find PythonInterp (missing: PYTHON_EXECUTABLE) 
CMake Error at build/_deps/cesiumnative-src/extern/draco/cmake/draco_emscripten.cmake:22 (message):
  Python required for Emscripten builds, but cmake cannot find it.
Call Stack (most recent call first):
  build/_deps/cesiumnative-src/extern/draco/cmake/draco_build_definitions.cmake:175 (draco_check_emscripten_environment)
  build/_deps/cesiumnative-src/extern/draco/CMakeLists.txt:84 (draco_set_build_definitions)

**Comment by josephbirkner** [comment]
@ashley-mort I summarized our solution here: https://github.com/CesiumGS/cesium/pull/11519#issuecomment-2025214844

That will probably be more helpful for you :rocket:

**Pull request merged**